### PR TITLE
Add naming test for ForbiddenVoid rule

### PR DESCRIPTION
**Pull request by schalkms**
This adds tests to reproduce issue #1642

### CODE REVIEW

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt [hunk 1]
```diff
@@ -37,6 +37,20 @@ class ForbiddenVoidSpec : Spek({
             assertThat(subject.compileAndLint(code)).isEmpty()
         }
 
+        it("does not report when functions or classes are called 'Void'") {
+            val code = """
+				class Void {
+                    fun void() {}
```
Comment (arturbosch) [inline detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: this should be fairly easy to fix, we just check if this name is a child of a declaration?

Comment (schalkms) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: ? What do you want to fix? There's nothing to fix here.

**Comment by schalkms** [comment]
I don't know what problem CodeFactor has. There's no trailing whitespace there.

File: detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt (hunk 1 above)
Comment (arturbosch) [reply detekt-rules/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/ForbiddenVoidSpec.kt]: Ah though this is a false positive but now seeing that the declaration name does not trigger it aka the isEmpty check :)

**Comment by arturbosch** [comment]
Not even in the testing code? It look wierd at github though.

**Pull request merged**